docs/proposals/upstream-settings.md

```diff
@@ -0,0 +1,225 @@
+
+This solution gives the object owners some knowledge that their object is affected by a policy but minimizes status updates by limiting them to when the affected object starts or stops being affected by a policy.
+
+The first step is adding the `gateway.nginx.org/UpstreamSettingsPolicyAffected: true` label to the affected Service. We also must set this Condition on all Routes that reference a Service affected by an `UpstreamSettingsPolicy`.
+Below is an example of what this Condition may look like:
+
+```yaml
+Conditions:
+  Type:                  gateway.nginx.org/UpstreamSettingsPolicyAffected
+  Message:               Object affected by an UpstreamSettingsPolicy.
+  Observed Generation:   1
+  Reason:                PolicyAffected
+  Status:                True
+```
+
+Implementing this involves defining a new Condition type and reason:
+
+```go
+package conditions
+
+import (
+    gatewayv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+const (
+    UpstreamSettingsPolicyAffected gatewayv1alpha2.PolicyConditionType = "gateway.nginx.org/UpstreamSettingsPolicyAffected"
+    PolicyAffectedReason gatewayv1alpha2.PolicyConditionReason = "PolicyAffected"
+)
+```
+
+Some additional rules:
+
+- This Condition and label should be added when the affected object starts being affected by an `UpstreamSettingsPolicy`.
+- When the last `UpstreamSettingsPolicy` affecting that object is removed, the Condition and label should be removed.
+- The Observed Generation is the generation of the affected object, not the generation of the `UpstreamSettingsPolicy`.
+
+## Use Cases
+
+- As an Application Developer, I want to be able to configure upstream settings for my application based on its behavior or requirements.
+  - I may have a large number of Pods for my Service and therefore need a larger memory zone for it.
+  - I may want to alter the keepalive settings for my upstream.
+
+## Testing
+
+- Unit tests
+- Functional tests that verify the attachment of the CRD to a Service, and that NGINX behaves properly based on the configuration.
+
+## Security Considerations
+
+Validating all fields in the `UpstreamSettingsPolicy` is critical to ensuring that the NGINX config generated by NGINX Gateway Fabric is correct and secure.
+
+All fields in the `UpstreamSettingsPolicy` will be validated with Open API Schema. If the Open API Schema validation rules are not sufficient, we will use [CEL](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules).
+
+RBAC via the Kubernetes API server will ensure that only authorized users can update the CRD.
+
+## Future Work
+
+- Add support for more [upstream-related directives](nginx-extensions.md#upstream-settings).
+- Add a `StreamUpstreamSettingsPolicy` that configures upstream settings for stream servers (TLSRoute, TCPRoute).
+
```

Comment:
Can we include a section on alternatives here? Having a record of the options we discussed over slack would be nice. I'm always forgetting things...